R/state.R

```diff
@@ -0,0 +1,22 @@
+check_pre_release_state = function(which, force = FALSE) {
+
+  # check version number
+  if (length(strsplit(as.character(desc::desc_get_version()), "[.]|-")[[1]]) < 4) {
+    ui_oops("pre_release(): Package versions needs to indicate a dev version before calling pre_release().")
+    state = FALSE
+    return(state)
+  } else if (file.exists("cran-comments.md")) {
+    ui_oops("pre_release(): cran-comments.md already exists.")
+    state = FALSE
+    return(state)
+  } else if (file.exists("CRAN-RELEASE")) {
+    ui_oops("pre_release(): CRAN-RELEASE already exists.")
+    state = FALSE
+    return(state)
+  } else if (git2r::is_branch(sprintf("cran-%s", update_version_helper(which)))) {
```

Comment:
`update_version_helper()` has side effects, I wouldn't expect this from a checker.

Reply:
Because `desc` is modified and avail in the GE? Isn't it recreated on every run?

We don't call `$write()`.

Or is it more about naming and we should rename to `update_version_helper_core()` or similar?